src/sleep/sleep.c

```diff
@@ -141,6 +146,38 @@ static int write_state(FILE **f, char **states) {
         return r;
 }
 
+static int configure_hibernation(void) {
+        _cleanup_free_ char *resume = NULL, *resume_offset = NULL;
+        int r;
+
+        /* check for proper hibernation configuration */
+        r = read_one_line_file("/sys/power/resume", &resume);
+        if (r < 0)
+                return log_debug_errno(r, "Error reading from /sys/power/resume: %m");
+
+        r = read_one_line_file("/sys/power/resume_offset", &resume_offset);
+        if (r < 0)
+                return log_debug_errno(r, "Error reading from /sys/power/resume_offset: %m");
+
+        if ((resume_offset && !streq(resume_offset, "0")) && (resume && !streq(resume, "0:0"))) {
+                log_debug("Hibernating using device id and offset read from /sys/power/resume: %s and /sys/power/resume_offset: %s", resume, resume_offset);
+                return 0;
+        }
+        else if (resume && !streq(resume, "0:0")) {
+                log_debug("Hibernating using device id read from /sys/power/resume: %s", resume);
+                return 0;
+        }
+        else if (resume_offset && !streq(resume_offset, "0"))
+                log_debug("Found offset in /sys/power/resume_offset: %s; no device id found in /sys/power/resume; ignoring offset", resume_offset);
+
+        /* if hibernation is not properly configured, attempt to calculate and write values */
+        r = write_hibernate_location_info();
+        if (r < 0)
+                return log_error_errno(r, "Failed to write hibernation location information: %m");
```

Comment:
if write_hibernate_location_info() fails, then we'll log twice here above LOG_DEBUG, as this function will and the calling function does again.

Please make sure that for each error we log exactly once above LOG_DEBUG level. (logging on debug level is generally always fine)